src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java

```diff
@@ -175,8 +177,12 @@ protected void configure(HttpSecurity http) throws Exception {
             .antMatchers("/websocket/**").permitAll()
             .antMatchers("/management/health").permitAll()
             .antMatchers("/management/info").permitAll()
-            // Only allow the configured IP address to access the prometheus endpoint, or allow 127.0.0.1 if none is specified
-            .antMatchers("/management/prometheus/**").hasIpAddress(monitoringIpAddress.orElse("127.0.0.1"))
+            // Only allow the configured IP addresses to access the prometheus endpoint
+            // This adds an access check like .access("hasIpAddress('127.0.0.1') or hasIpAddress('::1')")
+            .antMatchers("/management/prometheus/**").access(monitoringIpAddresses
+                .stream()
+                .map(ip -> String.format("hasIpAddress(\"%s\")", ip))
+                .collect(Collectors.joining(" or ")))
```

Comment:
```suggestion
            .antMatchers("/management/prometheus/**").access(getMonitoringAccessDefinition())
```

The builder for the access String could be moved to a small helper method.

```java
private String getMonitoringAccessDefinition() {
    // Only allow the configured IP addresses to access the prometheus endpoint
    // This adds an access check like .access("hasIpAddress('127.0.0.1') or hasIpAddress('::1')")
    return monitoringIpAddresses
                .stream()
                .map(ip -> String.format("hasIpAddress(\"%s\")", ip))
                .collect(Collectors.joining(" or "));
}
```

Reply:
I have moved it into a separate method, thanks!